src/models.py

```diff
@@ -27,26 +27,27 @@ def compile(element, compiler, **kw):
 
 Base = declarative_base()
 
-
-class Investment(Base):
+class Shares(Base):
     """
-    Our mighty investments have these columns
+    Here we store our shares bought and sold
     """
-    __tablename__ = "Investments"
+    __tablename__ = "Shares"
 
+    # Fill when bought
     id = Column(Integer, primary_key=True)
     post = Column(String(11), nullable=False)
     upvotes = Column(Integer, default=0)
-    comment = Column(String(11), nullable=False, unique=True)
-    name = Column(String(20), nullable=False, index=True)
-    amount = Column(BigInteger, default=100)
-    time = Column(Integer, server_default=unix_timestamp())
-    done = Column(Boolean, default=False, nullable=False)
-    response = Column(String(11))
-    final_upvotes = Column(Integer)
-    success = Column(Boolean, default=False)
-    profit = Column(BigInteger, default=0)
-
+    comment = Column(String(11), nullable=False, unique=True) # Comment that bought the share
+    name = Column(String(32), nullable=False, index=True)
+    initial_price = Column(BigInteger, default=-1) # The initial price of the stock
+    initial_shares = Column(BigInteger, default=-1) # Number of shares in the pool when bought
+    initial_time = Column(Integer, server_default=unix_timestamp()) # When bought
+    # Fill after selling
+    active = Column(Boolean, default=False) # If not sold and still owned by the user
```

Comment:
Interesting, so when the user sells, they must sell all of the shares bought in this row, they can't just sell a partial quantity? (This might be fine since it's simpler, just clarifying)

Reply:
No! When someone does let's say `!buy 10`, it does an integral computation (just a formula, aleardy integrated) to find the total price of all the shares. If one does not have enough to pay, reject the offer. If one does have the money, we will start allocating new shares of that submission one by one. So every share will have a lot of information sticked to it.